session/session.go

```diff
@@ -2230,6 +2232,18 @@ func (s *session) rollbackOnError(ctx context.Context) {
 	}
 }
 
+// CacheGeneralStmt parses the sql, generates the corresponding PlanCacheStmt and cache it.
+// The sql have to be parameterized, e.g. select * from t where a>?.
+func (s *session) CacheGeneralStmt(sql string) error {
+	if stmt := s.sessionVars.GetGeneralPlanCacheStmt(sql); stmt != nil {
```

Comment:
Why `stmt != nil` we should return `nil`? It seems if `stmt != nil`, it means there already had the `PlanCacheStmt` for the sql.

Reply:
Yes, if it's not nil we can skip the parse step so we can return directly in this case.